server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java

```diff
@@ -118,12 +118,11 @@ private void run() {
         TimeMetric timeMetric = metricFactory.timer(SPOOL_PROCESSING);
         try {
             processingActive.incrementAndGet();
-            return processMail(queueItem);
+            return processMail(queueItem)
+                .doOnSuccess(any -> timeMetric.stopAndPublish())
```

Comment:
you don't publish on error?

Reply:
it would be another metric because errors may have very different timings

Reply:
probably, but in the current state:
- the behavior is different than the previous one (which was always publishing and decrementing in finally)
- what happens to all these started metrics?

Reply:
Won't fix here, my intent is to fix current behaviour, not change the list of reported metrics.

(Please open a separate ticket)

Reply:
>  - what happens to all these started metrics?

They were IMMEDIATLY stopped, leading to think the processing took 0 ms

Reply:
Let's use doFinally